### PR TITLE
HPCC-13845 Improve the generated code for constant inline datasets

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
This change has two advantages:
* The rows are no longer cloned
* The generated code is much smaller

@jamienoss please review (and check differences in generated code)

**Comment by ghalliday** [comment]
Passes runtime regression suite.

File: rtl/include/eclhelper_base.hpp [hunk 1]
```diff
@@ -3721,6 +3721,33 @@ class CLibraryExistsAggregateArg : public CThorExistsAggregateArg
...
+class CLibraryConstantRawIteratorArg : public CThorLinkedRawIteratorArg
+{
+public:
+    inline CLibraryConstantRawIteratorArg(unsigned _numRows, byte * * _rows, IOutputMetaData * _meta)
+        : meta(_meta), numRows(_numRows), rows(_rows)
+    {
+        cur = 0;
+    }
+
+    virtual void onStart(const byte *, MemoryBuffer * ) { cur = 0U; }
```
Comment (jamienoss) [inline rtl/include/eclhelper_base.hpp]: Is it within the boundaries of our code practices to leave these parameters nameless?

Comment (richardkchapman) [inline rtl/include/eclhelper_base.hpp]: Recommended, if they are not used.

**Comment by jamienoss** [comment]
@richardkchapman good to merge.

**Pull request merged**